src/ray/core_worker/core_worker.cc

```diff
@@ -188,10 +227,120 @@ bool CoreWorker::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle) {
+Status CoreWorker::GetActorHandle(const ActorID &actor_id,
+                                  ActorHandle **actor_handle) const {
   auto it = actor_handles_.find(actor_id);
-  RAY_CHECK(it != actor_handles_.end());
-  return *it->second;
+  if (it == actor_handles_.end()) {
+    return Status::Invalid("Handle for actor does not exist");
+  }
+  *actor_handle = it->second.get();
+  return Status::OK();
+}
+
+Status CoreWorker::SubmitTask(const RayFunction &function,
+                              const std::vector<TaskArg> &args,
+                              const TaskOptions &task_options,
+                              std::vector<ObjectID> *return_ids) {
+  TaskSpecBuilder builder;
+  const int next_task_index = worker_context_.GetNextTaskIndex();
+  const auto task_id =
+      TaskID::ForNormalTask(worker_context_.GetCurrentJobID(),
+                            worker_context_.GetCurrentTaskID(), next_task_index);
+  BuildCommonTaskSpec(builder, worker_context_.GetCurrentJobID(), task_id,
+                      worker_context_.GetCurrentTaskID(), next_task_index, GetCallerId(),
+                      function, args, task_options.num_returns, task_options.resources,
+                      {}, TaskTransportType::RAYLET, return_ids);
+  return raylet_client_->SubmitTask(builder.Build());
+}
+
+Status CoreWorker::CreateActor(const RayFunction &function,
+                               const std::vector<TaskArg> &args,
+                               const ActorCreationOptions &actor_creation_options,
+                               ActorID *return_actor_id) {
+  const int next_task_index = worker_context_.GetNextTaskIndex();
+  const ActorID actor_id =
+      ActorID::Of(worker_context_.GetCurrentJobID(), worker_context_.GetCurrentTaskID(),
+                  next_task_index);
+  const TaskID actor_creation_task_id = TaskID::ForActorCreationTask(actor_id);
+  const JobID job_id = worker_context_.GetCurrentJobID();
+  std::vector<ObjectID> return_ids;
+  TaskSpecBuilder builder;
+  BuildCommonTaskSpec(
+      builder, job_id, actor_creation_task_id, worker_context_.GetCurrentTaskID(),
+      next_task_index, GetCallerId(), function, args, 1, actor_creation_options.resources,
+      actor_creation_options.placement_resources, TaskTransportType::RAYLET, &return_ids);
+  builder.SetActorCreationTaskSpec(actor_id, actor_creation_options.max_reconstructions,
+                                   actor_creation_options.dynamic_worker_options,
+                                   actor_creation_options.is_direct_call);
+
+  std::unique_ptr<ActorHandle> actor_handle(new ActorHandle(
+      actor_id, job_id, /*actor_cursor=*/return_ids[0], function.GetLanguage(),
+      actor_creation_options.is_direct_call, function.GetFunctionDescriptor()));
+  RAY_CHECK(AddActorHandle(std::move(actor_handle)))
+      << "Actor " << actor_id << " already exists";
+
+  auto status = raylet_client_->SubmitTask(builder.Build());
```

Comment:
maybe use `RAY_RETURN_NOT_OK` then set the return actor ID and return `Status::OK()` below just to keep things consistent